src/utils/validateMethod.js

```diff
@@ -0,0 +1,19 @@
+export const validateFactorialMethod = method => {
```

Comment:
I think these function along with the validateApproach.js can be unified into one function. Then the required list of method (loop, recursive, functional, oop, etc.) can be passed as a paremeter of the function.